projects/plugins/jetpack/_inc/client/newsletter/email-settings.jsx

```diff
@@ -17,6 +25,12 @@ const subscriptionsAndNewslettersSupportUrl =
 const FEATURED_IMAGE_IN_EMAIL_OPTION = 'wpcom_featured_image_in_email';
 const SUBSCRIPTION_EMAILS_USE_EXCERPT_OPTION = 'wpcom_subscription_emails_use_excerpt';
 const REPLY_TO_OPTION = 'jetpack_subscriptions_reply_to';
+const SENDER_NAME_OPTION = 'jetpack_subscriptions_sender_name';
+//Check for feature flag
+const urlParams = new URLSearchParams( window.location.search );
+const isNewsletterReplyToNameEnabled = urlParams.get( 'enable-newsletter-sender-name' ) === 'true';
+
+const { blogname } = window?.JP_CONNECTION_INITIAL_STATE ? window.JP_CONNECTION_INITIAL_STATE : {};
```

Comment:
Instead of relying on the Connection initial state, it may be best to rely on the Jetpack Dashboard initial state, defined here:
https://github.com/Automattic/jetpack/blob/8debc8539a172b79f0ec729d336be9d42824b791/projects/plugins/jetpack/_inc/lib/admin-pages/class-jetpack-redux-state-helper.php#L50

You'll find that the `blogname` option is already available there:
https://github.com/Automattic/jetpack/blob/8debc8539a172b79f0ec729d336be9d42824b791/projects/plugins/jetpack/_inc/lib/admin-pages/class-jetpack-redux-state-helper.php#L157

It's available for you to use via this reducer:
https://github.com/Automattic/jetpack/blob/d4b22e8d4b55f84c0348bd107c6d16549bd2227e/projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js#L73-L75

Relying on that will save you from having to make any changes to the Connection package this time around.